oci/layout/oci_archive_transport.go

```diff
@@ -0,0 +1,100 @@
+package layout
+
+import (
+	"github.com/containers/image/docker/reference"
+	"github.com/containers/image/image"
+	"github.com/containers/image/transports"
+	"github.com/containers/image/types"
+)
+
+func init() {
+	transports.Register(ArchiveTransport)
+}
+
+// ArchiveTransport is an ImageTransport for OCI archive
+// it creates an oci-archive tar file by calling into the OCI transport
+// tarring the directory created by oci and deleting the directory
+var ArchiveTransport = ociArchiveTransport{}
+
+type ociArchiveTransport struct{}
+
+var ociTr = ociTransport{}
+var ociRef types.ImageReference
+
+// ociArchiveReference is an ImageReference for OCI Archive paths
+// this is the same format as ociReference in oci_transport.go
+type ociArchiveReference struct {
+	dir         string
+	resolvedDir string
+	tag         string
+}
+
+func (t ociArchiveTransport) Name() string {
+	return "oci-archive"
+}
+
+// ParseReference converts a string, which should not start with the ImageTransport.Name prefix
+// into an ImageReference.
+func (t ociArchiveTransport) ParseReference(reference string) (types.ImageReference, error) {
+	ociRef, _ = ociTr.ParseReference(reference)
```

Comment:
The error return value needs to be handled.

Though, overall, it is unexpected to take the user-specified _file_ path and hand it off to a code which expects a _directory_ path; the behavior is really undefined.

This `ParseReference` should probably be doing its own parsing (though the code can look pretty similar, or even use a shared helper for parsing the reference/tag part) into a file+tag pair; then, something else (`NewImageDestination`?) should create a _temporary_ directory with a random name, and create an `oci/layout.ociReference` reference from that (directory, tag) part.